service/src/main/java/edu/softserve/zoo/service/impl/EmployeeServiceImpl.java

```diff
@@ -1,21 +1,68 @@
 package edu.softserve.zoo.service.impl;
 
+import edu.softserve.zoo.exceptions.ApplicationException;
+import edu.softserve.zoo.exceptions.NotFoundException;
 import edu.softserve.zoo.model.Employee;
 import edu.softserve.zoo.persistence.repository.EmployeeRepository;
 import edu.softserve.zoo.persistence.repository.Repository;
+import edu.softserve.zoo.persistence.specification.impl.employee.GetEmployeeByEmailSpecification;
 import edu.softserve.zoo.service.EmployeeService;
+import edu.softserve.zoo.service.exception.ServiceReason;
+import edu.softserve.zoo.util.Validator;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
+import java.util.List;
+import java.util.Locale;
 
+/**
+ * @author Ilya Doroshenko
+ */
 @Service
 public class EmployeeServiceImpl extends AbstractService<Employee> implements EmployeeService {
 
     @Autowired
     private EmployeeRepository employeeRepository;
 
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
     @Override
     Repository<Employee> getRepository() {
         return employeeRepository;
     }
+
+    @Override
+    @Transactional
+    public Employee save(Employee entity) {
+        String encodedPassword = passwordEncoder.encode(entity.getPassword());
+        String upperCaseEmail = entity.getEmail().toUpperCase(Locale.ROOT);
```

Comment:
i think both, because API consumer may use Postman, which won't uppercase anything.

CITEXT postgres data type solves that problem elegantly, it just makes "where" clause by that field case-insensitive. That fact frees us from any explicit case manipulations.